wpilib/wpilib/powerdistributionpanel.py

```diff
@@ -98,29 +97,13 @@ def clearStickyFaults(self):
         """
         hal.clearPDPStickyFaults(self.module)
 
-    # Live Window code, only does anything if live window is activated.
-
-    def getSmartDashboardType(self):
-        return "PowerDistributionPanel"
-
-    def initTable(self, subtable):
-        if subtable is not None:
-            self.chanEntry = []
-            for channel in range(16):
-                self.chanEntry.append(subtable.getEntry("Chan" + str(channel)))
-            self.voltageEntry = subtable.getEntry("Voltage")
-            self.totalCurrentEntry = subtable.getEntry("TotalCurrent")
-            self.updateTable()
-        else:
-            self.chanEntry = None
-            self.voltageEntry = None
-            self.totalCurrentEntry = None
-
-    def updateTable(self):
-        if self.chanEntry is not None:
-            for channel in range(16):
-                self.chanEntry[channel].setDouble(self.getCurrent(channel))
-        if self.voltageEntry is not None:
-            self.voltageEntry.setDouble(self.getVoltage())
-        if self.totalCurrentEntry is not None:
-            self.totalCurrentEntry.setDouble(self.getTotalCurrent())
+    def initSendable(self, builder):
+        builder.setSmartDashboardType("PowerDistributionPanel")
+        for chan in range(self.kPDPChannels):
+            builder.addDoubleProperty("Chan%s" % (chan,), self._getCurrentOf(chan), None)
```

Comment:
is there a more idiomatic way to do this in python?

Reply:
Doesn't this immediately call _getCurrentOf?  You want to pass it as a callable, so this should likely be a lambda instead.

Reply:
_getCurrentOf returns a lambda. python doesn't have final, so this is to bind the current value of chan to the lambda.

Reply:
yeah, that's probably as good as you're going to get. Worth adding a comment.

Reply:
There's always the `functools.partial` idiom for this. Otherwise I've seen `lambda x=x: ...` in the wild, but it reads weirdly to me.